packages/itwin/ec3-widget/src/components/CreateAssemblyComponent.tsx

```diff
@@ -0,0 +1,333 @@
+                              return { label: x, value: x };
+                            }) ?? []
+                          }
+                          value={assembly.materials.map((x) => x?.nameColumn)}
+                          onChange={(val, event) => {
+                            if (val) {
+                              onAssemblyDataChange(
+                                {
+                                  ...assembly,
+                                  materials:
+                                    event === "removed"
+                                      ? assembly.materials.filter((value) => val !== value?.nameColumn)
+                                      : [...(assembly.materials ?? []), { nameColumn: val }],
+                                },
+                                i,
+                              );
+                            }
+                          }}
+                          placeholder={
+                            isLoading
+                              ? "Loading elements"
+                              : assembly.reportTable === ""
+                                ? "Select report table first"
+                                : "Select property containing material names"
+                          }
+                          multiple
+                        />
+                      </>
+                    </ExpandableBlock>
+                  </>
+                );
+              })}
+            <div className="ec3w-button-row-above-stepper">
+              <Button
+                className="add-new-button"
+                styleType="borderless"
+                title="Add new"
+                startIcon={<SvgAdd />}
+                onClick={() => {
+                  // add new empty item to the assemblies array
+                  addNewEmptyAssembly();
+                }}
+              >
+                Add new
+              </Button>
+            </div>
+          </div>
+          <div className="ec3w-stepper-footer">
+            <Button onClick={props.onBackClick} className="ec3w-footer-button">
+              Back
+            </Button>
+            <Button
+              styleType="high-visibility"
+              className="ec3w-footer-button"
+              disabled={
+                allAssemblies === undefined ||
+                (allAssemblies && allAssemblies?.length === 0) ||
+                allAssemblies.some((assembly) => assembly.name === "") ||
+                allAssemblies.some((assembly) => assembly.name === undefined) ||
+                allAssemblies.some((assembly) => assembly.elementNameColumn === undefined) ||
+                allAssemblies.some((assembly) => assembly.elementQuantityColumn === undefined) ||
+                allAssemblies.some((assembly) => assembly.materials.length === 0) ||
+                allAssemblies.some((assembly) => assembly.reportTable === "")
+              }
```

Comment:
This looks somewhat inefficient.
```
const isNextDisabled = useMemo(() => 
  !allAssemblies || allAssemblies.length === 0 || allAssemblies.some(assembly => 
    !assembly.name ||
    !assembly.elementNameColumn || 
    !assembly.elementQuantityColumn || 
    assembly.materials.length === 0 || 
    !assembly.reportTable
  ),
  [allAssemblies]
);
```
Simplified it and wrapped it in a useMemo that way every rerender it doesn't run the expensive `some`. Let me know if this works.

Reply:
it works! Thanks